### PR TITLE
Enable FxCop Analyzers for the repo

### PR DESCRIPTION
**Pull request by Pilchie**
All rules are currently disabled, except for one that I enabled for
src/Http via a new ".editorconfig" file I added there.

Other changes:

* Allow editorconfigs in MVC and Razor to flow to the root
* Consolidate a few editorconfig settings
* Tweak Ruleset config in Azure/AzureAD where it clashed.

Addresses the beginning of #9620, but it's a fair chunk of work to
enable most rules through the whole repo. That can be done directory by
directory and rule by rule by dropping .editorconfig files though.

### CODE REVIEW

File: eng/Dependencies.props [hunk 1]
```diff
@@ -110,6 +110,7 @@ and are generated based on the last package release.
     <LatestPackageReference Include="Microsoft.AspNetCore.AzureAppServices.SiteExtension.3.1.x86" Version="$(MicrosoftAspNetCoreAzureAppServicesSiteExtension31PackageVersion)" />
     <LatestPackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.Runtime" Version="$(MicrosoftAspNetCoreComponentsWebAssemblyRuntimePackageVersion)" />
     <LatestPackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="$(MicrosoftBclAsyncInterfacesPackageVersion)" />
+    <LatestPackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="$(MicrosoftCodeAnalysisFxCopAnalyzersVersion)" />
```
Comment (dougbu) [inline eng/Dependencies.props:113]: It's apparent this package isn't in a feed we use. Could it be included in dotnet-eng or dotnet-tools @mmitche❔

Comment (Pilchie) [reply eng/Dependencies.props:113]: This is an OOB package, built apart from the stack.  It should probably just be excluded from source build.

**Review by dougbu — APPROVED** [review]
Hopefully just needs the `$(DotNetBuildFromSource)` check.

File: eng/Dependencies.props (hunk 1 above)
Comment (Pilchie) [reply eng/Dependencies.props:113]: Pushed the condition

**Comment by Pilchie** [comment]
@javiercn - I'll leave this in case anyone wants to discuss in the @dotnet/aspdoi meeting tomorrow (though I won't be there).

**Comment by javiercn** [comment]
LGTM, I'll make sure we discuss it on Today's meeting, but it sounds reasonable to me.

**Comment by javiercn** [comment]
We didn't get time to discuss this in today's meeting, but I think we can go ahead with these changes and start looking at individual rules.

**Review by javiercn — APPROVED** [review]

**Review by wtgodbe — APPROVED** [review]

**Pull request merged**